src/main/java/com/dreammaster/modhazardousitems/HazardousItemsHandler.java

```diff
@@ -399,32 +401,28 @@ private void CheckInventoryForItems(EntityPlayer pPlayer) {
         try {
             checkInventoryArray(pPlayer.inventory.mainInventory, pPlayer);
 
-            // M&B addition ------
             if (MineAndBladeBattleGear2.isModLoaded()) {
-                Class<?> c = pPlayer.inventory.getClass();
-                Field extraInv = null;
-                try {
-                    extraInv = c.getDeclaredField("battlegear2$extraItems");
-                } catch (NoSuchFieldException nsfe) {
-                    _mLogger.warn(
-                            "battlegear.changed.1",
-                            "Seems battlegear has updated/changed. Someone has to fix HazardousItems!");
-                }
-
-                if (extraInv == null) return;
-
                 try {
-                    extraInv.setAccessible(true);
-                    ItemStack[] tExtraInv = (ItemStack[]) extraInv.get(pPlayer.inventory);
+                    if (bg2ExtraInvField == null) {
+                        try {
+                            bg2ExtraInvField = pPlayer.inventory.getClass().getDeclaredField("battlegear2$extraItems");
+                        } catch (NoSuchFieldException nsfe) {
+                            _mLogger.warn(
+                                    "battlegear.changed.1",
+                                    "Seems battlegear has updated/changed. Someone has to fix HazardousItems!");
+                            bg2ExtraInvField = pPlayer.inventory.getClass().getDeclaredField("extraItems");
+                        }
+                        bg2ExtraInvField.setAccessible(true);
```

Comment:
This line should be inside `try` block. And there should be null check (see line 417).

Reply:
It is, see line 420

Reply:
getDeclaredField can't return null, it would a throw NoSuchFieldException, which is caught